src/praisonai/praisonai/async_agent_scheduler.py

```
@@ -183,18 +183,21 @@ async def stop(self) -> bool:
-                    await self._task
-                except asyncio.CancelledError:
-                    pass
-            
-        self.is_running = False
+                    await asyncio.wait_for(self._task, timeout=10)
+                except asyncio.TimeoutError:
+                    logger.warning("Scheduler task didn't stop gracefully, cancelling")
+                    self._task.cancel()
+                    try:
+                        await self._task
+                    except (asyncio.CancelledError, Exception) as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)